lib/tom_queue/deferred_work_manager.rb

```diff
@@ -113,6 +121,24 @@ def stop
       deferred_set && deferred_set.interrupt
     end
 
-  end
+    # Register handlers to shut down the manager gracefully
+    def register_signal_handlers
+      Thread.main[:signals] = []
+
+      [:QUIT, :INT, :TERM].each do |sig|
+        trap(sig) do
+          Thread.main[:signals] << sig
```

Comment:
Talked with @joshado on Slack. Having signal handling in the executable will definitely make things simple (as what we want). The only problem is that within the `trap` block `Mutex` and `Logger` (due to it's using mutex) can't be used, which might at some point might bite us when `DeferredWorkManager#stop` getting complex in the future.

 I’d prefer having a `CLI` class that deals with the signal handling and passed-in options stuff, and delegate it to `bin/deffer_manager` executable that stays in the tom queue (so we can see how far it goes).

Close the request for now.